sstables/sstables.cc

```diff
@@ -1389,7 +1389,7 @@ future<> sstable::read_filter(sstable_open_config cfg) {
 }
 
 void sstable::write_filter() {
-    if (!has_component(component_type::Filter)) {
+    if (!has_component(component_type::Filter) || _components->filter->memory_size() == 0) {
         return;
```

Comment:
When we write the TOC, will it reference the missing bloom filter?

Reply:
Why can't we reclaim memory from an unwritten filter? Say we have a particularly terrible estimate, should it be the first to be removed?

The logic should be self-consistent, if you skip writing due to this new condition, then the TOC writer should also consult it. Please audit all callers of has_component and leave comments.

Or ideally introduce new helpers (has_unevicted_filter()) and use them in the right places.

Reply:
Currently we track the total memory used by all bloom filters across all the sstables in the sstables_manager and reclaim from the bloom filter that occupies the most memory. We do not yet consider individual bloom filters sizes.

We could however do what you suggest as a first step of auto resizing bloom filters (#2024). If we find a bloom filter with a very bad estimate, skip writing it for now and in subsequent PRs add support to either fold or re generate filters from index.

Reply:
Yes all the tests pass without this change. I'll recheck the code once to confirm this is not needed and remove it from the patch.

Reply:
Please at least sprinkle comments, and follow up with a more robust API.

Reply:
I have removed this check from the fix.